tests/index.test.ts

```diff
@@ -0,0 +1,222 @@
+import { Pty } from '../wrapper';
+import { readdirSync, readlinkSync } from 'fs';
+import { describe, test, expect } from 'vitest';
+
+const EOT = '\x04';
+const procSelfFd = '/proc/self/fd/';
+
+type FdRecord = Record<string, string>;
+function getOpenFds(): FdRecord {
+  const fds: FdRecord = {};
+  if (process.platform !== 'linux') {
+    return fds;
+  }
+
+  for (const filename of readdirSync(procSelfFd)) {
+    try {
+      const linkTarget = readlinkSync(procSelfFd + filename);
+      if (linkTarget === 'anon_inode:[timerfd]' || linkTarget.startsWith("socket:[")) {
+        continue;
+      }
+
+      fds[filename] = linkTarget;
+    } catch (err: any) {
+      if (err.code === 'ENOENT') {
+        continue;
+      }
+      throw err;
+    }
+  }
+
+  return fds;
+}
+
+describe('PTY', () => {
+
+  test('spawns and exits', () => new Promise<void>(done => {
+    const oldFds = getOpenFds();
+    const message = 'hello from a pty';
+    let buffer = '';
+
+    const pty = new Pty({
+      command: '/bin/echo',
+      args: [message],
+      onExit: (err, exitCode) => {
+        expect(err).toBeNull();
+        expect(exitCode).toBe(0);
+        expect(buffer.trim()).toBe(message);
+        expect(getOpenFds()).toStrictEqual(oldFds);
+        done()
+      },
+    });
+
+    const readStream = pty.read;
+    readStream.on('data', (chunk) => {
+      buffer = chunk.toString();
+    });
+  }));
+
+  test('captures an exit code', () => new Promise<void>(done => {
+    const oldFds = getOpenFds();
+    const pty = new Pty({
+      command: '/bin/sh',
+      args: ['-c', 'exit 17'],
+      onExit: (err, exitCode) => {
+        expect(err).toBeNull();
+        expect(exitCode).toBe(17);
+        pty.close();
+        expect(getOpenFds()).toStrictEqual(oldFds);
+        done();
+      },
+    });
+  }));
+
+  test('can be written to', () => new Promise<void>(done => {
+    const oldFds = getOpenFds();
+
+    // The message should end in newline so that the EOT can signal that the input has ended and not
+    // just the line.
+    const message = 'hello cat\n';
+    let buffer = '';
+
+    // We have local echo enabled, so we'll read the message twice.
+    const result = process.platform === "darwin"
+      ? 'hello cat\r\n^D\b\bhello cat\r\n'
+      : 'hello cat\r\nhello cat\r\n';
+
+    const pty = new Pty({
+      command: '/bin/cat',
+      onExit: (err, exitCode) => {
+        expect(err).toBeNull();
+        expect(exitCode).toBe(0);
+        expect(buffer.trim()).toBe(result.trim());
+        expect(getOpenFds()).toStrictEqual(oldFds);
+        done();
+      },
+    });
+
+    const writeStream = pty.write;
+    const readStream = pty.read;
+
+    readStream.on('data', (data) => {
+      buffer += data.toString();
+    });
+    writeStream.write(message);
+    writeStream.end(EOT);
+  }));
+
+  test('can be resized', () => new Promise<void>(done => {
```

Comment:
one more test! can we assert that

`bash -c "for i in $(seq 0 1024); echo $i; done"`

returns the numbers between 0 and 1024, newline-delimited, inclusive? just to ensure that we're reading :100: .